packages/core/src/prompt.ts

```diff
@@ -590,10 +623,7 @@ export class Prompt<
 
       ctx.signal?.addEventListener('abort', streamAbort);
 
-      const metadata: TMetadata = {
-        ...(this.input.metadata || {} as TMetadata),
-        ...(await this.metadata(input, ctx) || {}),
-      };
+      const metadata = await (this.metadata as any)(input, ctx) as TMetadata;
```

Comment:
Don't use any - what's the deal here. I would prefer you switch back to the previous code here then use any.